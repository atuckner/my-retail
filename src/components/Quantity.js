import React, { PureComponent } from 'react';
import styled, { css } from 'styled-components';

const QuantityContainer = styled.div`
  height: 15px;
  border: 1px solid lightgray;
  border-radius: 2px;
  padding: 5px;
  width: 158px;
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
  margin-bottom: 15px;
`;

const QuantityText = styled.span`
  font-size: 12px;
  margin-right: 7px;
  margin-left: 7px;
`;

const QuantityLabel = styled.span`
  font-size: 11px;
  margin-right: 7px;
  margin-left: 3px;
`;

const circleButtonStyles = css`
  width: 18px;
  height: 18px;
  border-radius: 100%;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  cursor: pointer;
  background: #aaaaaa4f;
  ::before,
  ::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
`;

const CounterWrapper = styled.div`
  margin-right: 5px;
  margin-top: -3px; 
`;

const Plus = styled.div`
  ${circleButtonStyles}
  ::before,
  ::after {
    background: white;
    box-shadow: 1px 1px 1px #ffffff9e;
  }
  ::before {
    width: 2px;
    margin: 4px auto;
  }
  ::after {
    margin: auto 3px;
    height: 2px;
    box-shadow: none;
  }
`;

const Minus = styled.div`
  ${circleButtonStyles}
  ::before {
    background: white;
    margin: auto 3px;
    height: 2px;
    box-shadow: 0px 1px 1px #ffffff9e;
  }
`;

export default class Quantity extends PureComponent {
  state = {
    quantity: 1
  }

  incrementQuantity = () => {
    this.setState(prevState => ({
      quantity: prevState.quantity + 1
    }))
  }

  decrementQuantity = () => {
    if (this.state.quantity === 1) return;

    this.setState(prevState => ({
      quantity: prevState.quantity - 1
    }))
  }

  render() {
    const { quantity } = this.state;

    return (
      <QuantityContainer>
        <QuantityLabel>quantity:</QuantityLabel>
        <CounterWrapper>
          <Minus onClick={() => { this.decrementQuantity() }}></Minus>
          <QuantityText>
            {quantity}
          </QuantityText>
          <Plus onClick={() => { this.incrementQuantity() }}></Plus>
        </CounterWrapper>
      </QuantityContainer>
    );
  }
}
