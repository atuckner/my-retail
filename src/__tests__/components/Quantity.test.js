import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'

import Quantity from '../../components/Quantity';

test('render Quantity', () => {
  const component = renderer.create(
    <Quantity/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
