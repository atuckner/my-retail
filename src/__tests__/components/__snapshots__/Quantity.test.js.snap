// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Quantity 1`] = `
.c0 {
  height: 15px;
  border: 1px solid lightgray;
  border-radius: 2px;
  padding: 5px;
  width: 158px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 10px;
  margin-bottom: 15px;
}

.c4 {
  font-size: 12px;
  margin-right: 7px;
  margin-left: 7px;
}

.c1 {
  font-size: 11px;
  margin-right: 7px;
  margin-left: 3px;
}

.c2 {
  margin-right: 5px;
  margin-top: -3px;
}

.c5 {
  width: 18px;
  height: 18px;
  border-radius: 100%;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  cursor: pointer;
  background: #aaaaaa4f;
}

.c5::before,
.c5::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c5::before,
.c5::after {
  background: white;
  box-shadow: 1px 1px 1px #ffffff9e;
}

.c5::before {
  width: 2px;
  margin: 4px auto;
}

.c5::after {
  margin: auto 3px;
  height: 2px;
  box-shadow: none;
}

.c3 {
  width: 18px;
  height: 18px;
  border-radius: 100%;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  cursor: pointer;
  background: #aaaaaa4f;
}

.c3::before,
.c3::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c3::before {
  background: white;
  margin: auto 3px;
  height: 2px;
  box-shadow: 0px 1px 1px #ffffff9e;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    quantity:
  </span>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    />
    <span
      className="c4"
    >
      1
    </span>
    <div
      className="c5"
      onClick={[Function]}
    />
  </div>
</div>
`;
